subprojects/core/src/main/java/org/gradle/initialization/InternalBuildFinishedListener.java

```diff
@@ -25,8 +25,9 @@
      * Called after all user buildFinished hooks have been executed, but before
      * services are shutdown.
      * @param gradle the Gradle instance being finalized
+     * @param failed
      */
-    default void buildFinished(GradleInternal gradle) {
+    default void buildFinished(GradleInternal gradle, boolean failed) {
```

Comment:
Isn't this part of the public API? Isn't this a breaking change?

Reply:
Even though the package name does not contain `internal`, this was not reported by the binary compatibility check. So we are good. And the class name is explicit 😉

Reply:
Consider moving this to an `internal` package then potentially?